wasm/lib/src/browser_module.rs

```diff
@@ -270,6 +263,65 @@ fn promise_catch(vm: &VirtualMachine, args: PyFuncArgs) -> PyResult {
     Ok(PyPromise::new_obj(promise_type, ret_promise))
 }
 
+#[derive(Debug)]
+struct PyDocument {
+    doc: web_sys::Document,
+}
+type PyDocumentRef = PyRef<PyDocument>;
+
+impl PyValue for PyDocument {
+    fn class(vm: &VirtualMachine) -> PyObjectRef {
+        vm.class("browser", "Document")
+    }
+}
+
+fn document_query(zelf: PyDocumentRef, query: PyStringRef, vm: &VirtualMachine) -> PyResult {
+    let elem = zelf
+        .doc
+        .query_selector(&query.value)
+        .map_err(|err| convert::js_py_typeerror(vm, err))?;
+    let elem = match elem {
+        Some(elem) => PyElement { elem }.into_ref(vm).into_object(),
+        None => vm.get_none(),
+    };
+    Ok(elem)
+}
+
+#[derive(Debug)]
+struct PyElement {
+    elem: web_sys::Element,
+}
+type PyElementRef = PyRef<PyElement>;
+
+impl PyValue for PyElement {
+    fn class(vm: &VirtualMachine) -> PyObjectRef {
+        vm.class("browser", "Element")
+    }
+}
+
+fn elem_get_attr(
```

Comment:
I know it won't let you do `impl PyElementRef`, but how about `impl PyElement`? In the future we'll be able to make it follow the pattern once `arbitrary-self-types` are allowed.

Reply:
Oh, I didn't consider that. Sounds good!